[Yap Wei Jie Joshua] iP

joshuayapwj98 opened the pull request:
# Harman/Duke
> "Talent wins games, but teamwork and intelligence wins championships." ― Michael Jordan

DukePro frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
~FAST~ SUPER FAST to use

All you need to do is,

download it from [here](https://github.com/e0544333/ip/releases/tag/A-Jar).
double-click it.
Add your tasks.
let it manage your errands for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can also use it to practice Java. Here's the primary method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

joshuayapwj98 committed:
Greet, Echo, Exit

joshuayapwj98 committed:
Add, List

joshuayapwj98 committed:
Mark as Done

joshuayapwj98 committed:
ToDos, Events, Deadlines

joshuayapwj98 committed:
Add automation-TextUI Testing

joshuayapwj98 committed:
Fix space character user input

joshuayapwj98 committed:
Fix unspecified event or deadline information

joshuayapwj98 committed:
Refactor logic statement for deadline and event

joshuayapwj98 committed:
Fix Deadline and Event output to display pre-text

joshuayapwj98 committed:
Add delete function

* Added a helper list to print out all the different functions made available by Duke bot.

joshuayapwj98 committed:
Add Enum to handle different commands

joshuayapwj98 committed:
Fix mark number bug

joshuayapwj98 committed:
Change task unmark and mark method behavior

joshuayapwj98 committed:
Add IO feature to save file into root directory

joshuayapwj98 committed:
Add LocalDateTime feature for deadlines

joshuayapwj98 committed:
Merge branch 'branch-Level-7'

joshuayapwj98 committed:
Merge branch 'branch-Level-8'

- Removed import java.io.* and added specific imports

joshuayapwj98 committed:
Refactor program to follow OOP Structure

TODO:
- Test more edge cases that could break the program.

joshuayapwj98 committed:
Package files

joshuayapwj98 committed:
Fix error message

joshuayapwj98 committed:
Add JUnit testing

- Refactored some functions to cater for edge cases

joshuayapwj98 committed:
Add code documentation

joshuayapwj98 committed:
Add Find feature

joshuayapwj98 committed:
Change to follow Coding Standard

ryantanlien commented on src/main/java/command/EventCommand.java:
```diff
@@ -0,0 +1,50 @@
+            ui.showTaskMessage(event);
+            ui.printTasksCountMessage(tasks);
+        } catch (DukeException e) {
+            // e.printStackTrace();
```
Maybe it would be a good idea to remove unused code? 😆

ryantanlien commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,43 @@
+
+    protected String words;
+
+    /**
```
I know it's really self explanatory but this probably needs a header? 🤷

ryantanlien commented on src/main/java/task/Todo.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    /**
+     * Returns todo details.
```
Maybe todo can be capitalized to ToDo to show that it returns a instance of a class?

ryantanlien reviewed (commented):
All in all the code standard is great! Really what I'm doing here is just nitpicking haha! Good job 👍

joshuayapwj98 committed:
Merge branch 'add-gradle-support'

joshuayapwj98 committed:
Add Gradle support

joshuayapwj98 committed:
Add Checkstyle to project file

- Gradle to use custom checkstyle for code style.

joshuayapwj98 committed:
First iteration of GUI

- Added controllers into the project

joshuayapwj98 committed:
Improve using fxml

joshuayapwj98 committed:
Add support for Duke in the bot

TODO:
- Fix the label to display the list of tasks

joshuayapwj98 committed:
Merge branch 'GUI'

joshuayapwj98 committed:
Add java varags in Response class

joshuayapwj98 committed:
Use checkstyle in config folder

joshuayapwj98 committed:
Major changes

- Change javaFX version to 17
- Fix delete command bug where no text is being displayed
- Fix ExitCommand

joshuayapwj98 committed:
Merge branch 'GUI'

rcjj98 commented on src/main/java/Duke.java:
```diff
@@ -69,7 +71,7 @@ public static void main(String[] args) throws IOException {
                 printLines();
             }
-        }
+        } while(response.compareTo("bye") != 0);
```
just nitpicking, reserved words (e.g. "while") should be followed with white spaces

rcjj98 commented on src/main/java/Deadline.java:
```diff
@@ -4,7 +4,11 @@
+        this.by = by.trim();
+    }
+
+    public boolean checkDateline() {
```
just a small coding standard violation, names representing methods should follow camelCase style, (i.e the method should be named "checkDateLine" instead of "checkDateline"

rcjj98 reviewed (commented):
LGTM. Just a few nits to fix like minor naming convention violations and missing white spaces after reserved words

joshuayapwj98 committed:
Change branch name

joshuayapwj98 committed:
Merge branch 'GUI'

joshuayapwj98 committed:
Change branch name

joey-chance commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,61 @@
+
+    private static boolean isExit = false;
+    protected Response response;
+    /**
```
Shouldn't there be a line separating the class variables from the method or its javadocs? 🤔

joey-chance commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,61 @@
+    public Command() {
+    }
+
+    /**
```
Shouldn't there be no line separating the class methods from other methods? 🤔  I noticed the same issue in several other places too.

joey-chance commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+
+        if (this.index > 0 && this.index <= tasks.size()) {
```
Shouldn't there be no line separating the method declaration and the first line of the body? 🤔

joey-chance commented on src/main/java/controller/DialogBox.java:
```diff
@@ -0,0 +1,79 @@
+
+import java.io.IOException;
+import java.util.Collections;
+
```
Any reason why you added a line break after the first two import statements?

joey-chance reviewed (commented):
LGTM. 👍  Just a few nits to fix here and there.

joshuayapwj98 committed:
Fix ui command display

joshuayapwj98 committed:
Add assertions

joshuayapwj98 committed:
Amend code to meet quality check

Command classes: print error handling stack trace message

Command abstract method execute throw DukeException. While each different Command implementation (e.g. ToDo, Event, Deadline) print error stack trace.
 
The uncommon behaviors across command method execute cause unexpected behaviours when encounter error.

Standardising the  exception handling behavior enables us to catch all common exceptions and converges error handling into a single try-catch block on the main method call.

Let's ensure all common exceptions are thrown and is encapsulated by a DukeException error object. In addition, assert presumable variables.

Assert ensures unexpected code behaviour is not reached.

joshuayapwj98 committed:
Add java stream function

TaskList method: use for loop to extract tasks by search term

TaskList find method creates a new ArrayList and use a for-loop to iterate through each element to find the task(s) that contains the search term.
 
Using a for-loop is more error-prone. Working with Streams enable us with higher level of abstraction. In addition, it generates more intuitive code.

Let's convert the method to implement Java's stream feature to streamline the search process.

Removes the need to perform multiple if-else checks by using a filter function and maps it a List collection.

Refer to this post on the advantages of using streams
https://www.capitalone.com/tech/software-engineering/java-streams-explained-simple-example/

joshuayapwj98 committed:
Merge pull request #1 from e0544333/branch-A-Assertions

Add assertions

joshuayapwj98 committed:
Merge pull request #2 from e0544333/branch-A-CodeQuality

Amend code to meet quality check

joshuayapwj98 committed:
Merge pull request #3 from e0544333/branch-A-Streams

Add java stream function

joshuayapwj98 committed:
Add Github Actions

joshuayapwj98 committed:
Merge branch 'branch-Level-10'

joshuayapwj98 committed:
Merge pull request #4 from e0544333/e0544333-patch-1

Add Github Actions

joshuayapwj98 committed:
Add basic priority feature

- Added basic functionality of allowing users to add a priority to each task.

TODO:
- Fix the content that is being split in Parser class.
- Display a priority for the user.

joshuayapwj98 committed:
Add Priorities Enum 

- Added Priorities enum to facilitate the priority rankings

joshuayapwj98 committed:
Add UI text to support user experience

joshuayapwj98 committed:
Merge branch 'BCD-Extension'

joshuayapwj98 committed:
Change syntax of invalid range message

joshuayapwj98 committed:
Add new images for user and bot

joshuayapwj98 committed:
Add new feature on command list

joshuayapwj98 committed:
Add initial UG

joshuayapwj98 committed:
Merge tag 'A-UserGuide'

joshuayapwj98 committed:
Update README.md

joshuayapwj98 committed:
Merge remote-tracking branch 'origin/master'

joshuayapwj98 committed:
Merge tag 'A-BetterGui'

joshuayapwj98 committed:
Add Ui screenshot

joshuayapwj98 committed:
Merge branch 'A-UserGuide'

joshuayapwj98 committed:
Fix issues from iP code quality feedback
